larsim/PhotonPropagation/PhotonVisibilityService.cc

```diff
@@ -831,4 +916,88 @@ namespace phot {
   }
 
+  void PhotonVisibilityService::findVoxelSuggestion(std::string& logString,
+                                                    float tpcMin,
+                                                    float tpcMax,
+                                                    float cryoMin,
+                                                    float cryoMax,
+                                                    int& nVoxels,
+                                                    float& voxelMin,
+                                                    float& voxelMax,
+                                                    float voxelSizeGoal) const
+  {
+    // Scan the "jog" parameter, which adds an integer amount of voxels within the TPC
+    int bestJog = 0;
+    float bestResult = 1000;
+    std::string dummyString = "";
+    for (int jog = -10; jog <= 10; ++jog) {
+      dummyString = "";
+      float result = testVoxelSuggestion(dummyString,
+                                         tpcMin,
+                                         tpcMax,
+                                         cryoMin,
+                                         cryoMax,
+                                         nVoxels,
+                                         voxelMin,
+                                         voxelMax,
+                                         voxelSizeGoal,
+                                         jog);
+      if (result < bestResult) {
+        bestResult = result;
+        bestJog = jog;
+      }
+    }
+
+    // Print the best result (and update the outputs)
+    testVoxelSuggestion(logString,
+                        tpcMin,
+                        tpcMax,
+                        cryoMin,
+                        cryoMax,
+                        nVoxels,
+                        voxelMin,
+                        voxelMax,
+                        voxelSizeGoal,
+                        bestJog);
+  }
+
+  float PhotonVisibilityService::testVoxelSuggestion(std::string& logString,
+                                                     float tpcMin,
+                                                     float tpcMax,
+                                                     float cryoMin,
+                                                     float cryoMax,
+                                                     int& nVoxels,
+                                                     float& voxelMin,
+                                                     float& voxelMax,
+                                                     float voxelSizeGoal,
+                                                     int jog) const
+  {
+    // Examine TPC to establish voxel size
+    float tpcSize = tpcMax - tpcMin;
+    int tpcVoxelNumber = ceil(tpcSize / voxelSizeGoal) + jog; // units are cm
+    float voxelSize = tpcSize / tpcVoxelNumber;
+    float cryoSize = cryoMax - cryoMin;
+    logString += "\tTPC size " + std::to_string(tpcSize) + "cm requires " +
```

Comment:
Assuming you change the `testVoxelSuggestion(...)` function signature as suggested, you can then do:

```c++
if (logString) {
  *logString += ...
}

Reply:
Agreed, done